Insert CSV with --detect-types crashes for CSV files that contain only a header row

I exported an empty table to csv, so that the csv file contains only the header row. When trying to create an SQLite DB file from that, using
`sqlite-utils insert "$dbfile" "$table" "$csvfile" --csv --detect-types`
I run into this error:
  File "/root/.local/pipx/venvs/sqlite-utils/lib/python3.11/site-packages/sqlite_utils/cli.py", line 1244, in insert
    insert_upsert_implementation(
  File "/root/.local/pipx/venvs/sqlite-utils/lib/python3.11/site-packages/sqlite_utils/cli.py", line 1128, in insert_upsert_implementation
    db[table].transform(types=tracker.types)
  File "/root/.local/pipx/venvs/sqlite-utils/lib/python3.11/site-packages/sqlite_utils/db.py", line 1780, in transform
    assert self.exists(), "Cannot transform a table that doesn't exist yet"
AssertionError: Cannot transform a table that doesn't exist yet

If I run it without --detect-types, then the issue naturally does not occur.
If --detect-types is going to be the default in the future, then I would suggest fixing this, for example by making the following change in cli.py on line 1179:
`if tracker is not None and db.table(table).exists():`

--- a/sqlite_utils/cli.py
+++ b/sqlite_utils/cli.py
@@ -1176,7 +1176,7 @@ def insert_upsert_implementation(
                 )
             else:
                 raise
-        if tracker is not None:
+        if tracker is not None and db.table(table).exists():
             db.table(table).transform(types=tracker.types)
 
         # Clean up open file-like objects
@@ -2033,7 +2033,7 @@ def memory(
                 rows = (_flatten(row) for row in rows)
 
             db.table(file_table).insert_all(rows, alter=True)
-            if tracker is not None:
+            if tracker is not None and db.table(file_table).exists():
                 db.table(file_table).transform(types=tracker.types)
             # Add convenient t / t1 / t2 views
             view_names = ["t{}".format(i + 1)]

--- a/tests/test_cli_insert.py
+++ b/tests/test_cli_insert.py
@@ -597,3 +597,21 @@ def test_insert_streaming_batch_size_1(db_path):
     proc.stdin.close()
     proc.wait()
     assert proc.returncode == 0
+
+
+def test_insert_csv_headers_only(tmpdir):
+    """Test that CSV with only header row (no data) works with --detect-types (issue #702)"""
+    db_path = str(tmpdir / "test.db")
+    csv_path = str(tmpdir / "headers_only.csv")
+    with open(csv_path, "w") as fp:
+        fp.write("id,name,age\n")
+    # Should not crash with --detect-types (which is now the default)
+    result = CliRunner().invoke(
+        cli.cli,
+        ["insert", db_path, "data", csv_path, "--csv"],
+        catch_exceptions=False,
+    )
+    assert result.exit_code == 0
+    # Table should not exist since there were no data rows
+    db = Database(db_path)
+    assert not db["data"].exists()

